arbor/local_alloc.cpp

```diff
@@ -1,13 +1,14 @@
 #include <arbor/domain_decomposition.hpp>
-#include <arbor/threadinfo.hpp>
+#include <arbor/execution_context.hpp>
+#include <threading/threading.hpp>
 
 #include "hardware/node_info.hpp"
 
 namespace arb {
 
-proc_allocation local_allocation() {
+proc_allocation local_allocation(execution_context* ctx) {
```

Comment:
Unless we have a good reason to use pass-by-pointer, which tends to obscure the lifetime requirements when used in a function interface, I'd recommend defaulting to pass-by-value or pass-by-const-reference. Presuming that we keep `execution_context` as essentially a bag of shared pointers, I'd prefer to use pass-by-value when it's going to be copied and used in some stateful object, and pass-by-const-reference otherwise where it makes sense.